library/core/class.dispatcher.php

```diff
@@ -796,6 +797,13 @@ private function _mapParts($Parts, $ControllerKey) {
             }
         }
     }
+
+    /**
+     * Trigger an event allowing addons to modify response headers.
+     */
+    public function sendHeaders() {
```

Comment:
In PHP, `$this` is [automatically bound](http://php.net/manual/en/functions.anonymous.php#example-199) to the function when you create an anonymous function in the context of a class.  We don't need to go the route of setting `$that` to `$this`.

The anonymous function suggestion is solid and has been implemented.